src/lib/utils/options/declaration.ts

```diff
@@ -92,11 +92,11 @@ export class OptionDeclaration {
                 }
                 break;
             case ParameterType.Map:
-                if (this.map !== 'object') {
-                    const key = value ? (value + '').toLowerCase() : '';
-                    if (key in this.map) {
-                        value = this.map[key];
-                    } else if (errorCallback) {
+                const key = value ? (value + '').toLowerCase() : '';
+                try {
+                    this.getMapValue(key);
+                } catch (error) {
```

Comment:
Can we please avoid using `try..catch` statements here. It seems like we should be able to keep most of the code the same as it was instead of changing all this.